apm-agent-core/src/test/java/co/elastic/apm/agent/impl/MetaDataTest.java

```diff
@@ -92,7 +93,10 @@ void testCloudProvider_SingleProvider(CoreConfiguration.CloudProvider provider)
         }
         if (provider != GCP && provider != currentCloudProvider) {
             // The GCP fails quickly on DNS lookup
-            assertThat(timeoutException).isInstanceOf(TimeoutException.class);
+            // [windows10] With AWS provider TimeoutException not captured
+            if (!(System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("win") && provider == AWS)) {
+                assertThat(timeoutException).isInstanceOf(TimeoutException.class);
+            }
```

Comment:
Not sure this is required, I think it's enough to check `assertThat(timeoutException).isNotNull()`

Reply:
This test is flakky, sometimes `TimeoutException` is thrown sometimes not

Reply:
Sorry, what I meant is changing the `catch` to catch any exception type and only assert it is not `null`, without caring about the actual type.

Reply:
OK, then let's just remove the assertion on timeout exceptions in these tests. You should still `get()` the metadata with a timeout millisecond argument, but go straight to using the safe one. The `verifyMetaData()` method will take care of verifying we only get a cloud provider where we expect it.